Extend REFRESH statement docs (#10227)
crate/crate: docs/sql/statements/refresh.rst

Description
===========

The REFRESH TABLE command refreshes one or more tables, making all changes made
to that table visible to subsequent commands.

The ``PARTITION`` clause can be used to refresh specific partitions of a
partitioned table instead of all partitions. The ``PARTITION`` clause requires
all partitioned columns with the values that identify the partition as
argument.

In case the ``PARTITION`` clause is omitted all open partitions will be
refreshed. Closed partitions are not refreshed.

See :ref:`partitioned_tables` for more information on partitioned tables.

For performance reasons, refreshing tables or partitions should be avoided as
it is an expensive operation. By default CrateDB periodically refreshes the
tables anyway. See :ref:`refresh_data` and :ref:`sql_ref_refresh_interval` for
more information about the periodic refreshes.

Without an explicit ``REFRESH``, other statements like ``UPDATE``, ``DELETE``
or ``SELECT`` won't see data until the periodic refresh happens.

An exception to that are statements which can filter on a primary key with an

> exact match on one or more values. For example, looking up a single document in

Reviewer: if i got this right all primary key values have to be addressed, right?
```suggestion
exact match on all primary key values. For example, looking up a single document in
```

Author: It was a bit amgigious. In the context of a single record all primary key values must be present.
What I wanted to say is that you can lookup multiple records - but I think the later example shows that anyhow, so I used your suggestion and tweaked it a bit further

a table with a single primary key column::

    WHERE pk = 'ID1'

If the primary key consists of multiple columns it would look like this::

    WHERE pk1 = 'ID_PART_1' AND pk2 = 'ID_PART_2'

Or if you want to query multiple records::

>     WHERE pk = 'ID1' OR pk = 'ID2' OR PK = 'ID3'

Author: As you mentioned - no that won't result in a primary key lookup. Not sure if it's necessary to include counter examples like that - given that the text tells users how to answer questions like that (and you already proofed that it's easy enough to do :) )
